pkg/commands/root.go

```diff
@@ -0,0 +1,75 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package commands
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+
+	"github.com/spf13/cobra"
+)
+
+type CommonOptions struct {
+	Debug    bool
+	Log      *log.Logger
+	DebugLog *log.Logger
+}
+
+func ShowHelp(cmd *cobra.Command, args []string) error {
+	fmt.Fprint(cmd.OutOrStderr(), cmd.UsageString())
+	return nil
+}
+
+type NewCommandFunc func(ko *CommonOptions) *cobra.Command
+
+// NewRootCommand returns entrypoint command to interact with all other commands
+func NewRootCommand(extraCmds ...NewCommandFunc) *cobra.Command {
```

Comment:
I don't think we will need `extraCmds`. We added this in `deployer` to make as easy as possible to consume the `command` package for third parties, but I don't think we have this usecase just yet (or on the radar in general AFAIK).